v201710/sandbox_test.go

```diff
@@ -934,30 +936,24 @@ func TestRateError(t *testing.T) {
 
 }
 
-func TestSandboxEmptyErrorMessage(t *testing.T) {
-	config := getTestConfig()
-	campaigns, _, err := NewCampaignService(&config.Auth).Get(Selector{
-		Fields: []string{"Id", "Name", "CampaignId"},
-	})
+type StringClient string
 
-	if err != nil {
-		t.Fatal(err)
+func (s StringClient) Do(req *http.Request) (*http.Response, error) {
+	return &http.Response{
+		Body:       BufferCloser{bytes.NewBufferString(string(s))},
+		StatusCode: 500,
+	}, nil
+}
+
+func TestSandboxEmptyErrorMessage(t *testing.T) {
+	client := StringClient(`<soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/"><soap:Body><soap:Fault><faultcode>soap:Client</faultcode><faultstring>Unmarshalling Error: cvc-complex-type.2.4.a: Invalid content was found starting with element 'adServingOptimizationStatus'. One of '{"https://adwords.google.com/api/adwords/cm/v201710":status, "https://adwords.google.com/api/adwords/cm/v201710":settings, "https://adwords.google.com/api/adwords/cm/v201710":labels, "https://adwords.google.com/api/adwords/cm/v201710":forwardCompatibilityMap, "https://adwords.google.com/api/adwords/cm/v201710":biddingStrategyConfiguration, "https://adwords.google.com/api/adwords/cm/v201710":contentBidCriterionTypeGroup, "https://adwords.google.com/api/adwords/cm/v201710":baseCampaignId, "https://adwords.google.com/api/adwords/cm/v201710":baseAdGroupId, "https://adwords.google.com/api/adwords/cm/v201710":trackingUrlTemplate, "https://adwords.google.com/api/adwords/cm/v201710":urlCustomParameters, "https://adwords.google.com/api/adwords/cm/v201710":adGroupType, "https://adwords.google.com/api/adwords/cm/v201710":adGroupAdRotationMode}' is expected. </faultstring></soap:Fault></soap:Body></soap:Envelope>`)
```

Comment:
What's going on here?

Reply:
The issue originally was that adwords sometimes gives a Non ApiExceptionFault. Gads was not taking it into account and giving an err with a blank message. The actual change was to handle such cases in base.go with the err message populated. In the test case, putting the sample response from adwords with a mock client so that we can take such Fault responses into account.